+hw/DaqController.m

```diff
@@ -193,28 +246,36 @@ function command(obj, varargin)
     
     function set.Value(obj, v)
       readyWait(obj);
-      obj.DaqSession.outputSingleScan(v(obj.AnalogueChannelsIdx));
+      if any (obj.AnalogueChannelsIdx)
+        obj.DaqSession.outputSingleScan(v(obj.AnalogueChannelsIdx));
+      end
       if any(~obj.AnalogueChannelsIdx)
         obj.DigitalDaqSession.outputSingleScan(v(~obj.AnalogueChannelsIdx));
       end
       obj.CurrValue = v;
     end
     
     function reset(obj)
-      stop(obj.DaqSession);
+      if ~isempty(obj.DaqSession)
+        stop(obj.DaqSession);
+      end
       if ~isempty(obj.DigitalDaqSession)
         stop(obj.DigitalDaqSession);
       end
       v = [obj.SignalGenerators.DefaultValue];
-      outputSingleScan(obj.DaqSession, v(obj.AnalogueChannelsIdx));
+      if any(obj.AnalogueChannelsIdx)
+        outputSingleScan(obj.DaqSession, v(obj.AnalogueChannelsIdx));
+      end
       if any(~obj.AnalogueChannelsIdx)
         outputSingleScan(obj.DigitalDaqSession, v(~obj.AnalogueChannelsIdx));
```

Comment:
Can you outputSingleScan on a clocked channel? Same for line 253

Reply:
I think you want to set a different property or something for a clocked session...

Reply:
yes, outputSingleScan does work for counter channels